impl/src/main/java/io/jsonwebtoken/impl/DefaultJwtParser.java

```diff
@@ -107,13 +108,6 @@ public DefaultJwtParser() { }
         this.base64UrlDecoder = base64UrlDecoder;
         this.deserializer = deserializer;
         this.compressionCodecResolver = compressionCodecResolver;
-
-        if (this.deserializer == null) {
-            //try to find one based on the runtime environment:
-            InstanceLocator<Deserializer<Map<String, ?>>> locator =
-                Classes.newInstance("io.jsonwebtoken.impl.io.RuntimeClasspathDeserializerLocator");
-            this.deserializer = locator.getInstance();
-        }
```

Comment:
NOTE: this constructor was after the last release, so moving this block to the DefaultJwtParser builder is _more_ correct

Reply:
Seeing this elucidates something that the ParserBuilder is going to fix for applications:

In existing releases (before 0.11.0), if a `Deserializer` implementation wasn't found, the exception would be thrown when `parse` is called.  That means this exception wouldn't be discovered until well after the application has started - e.g. during an HTTP request.  Since the default Deserializer implementation lookup isn't based on runtime (e.g. request) input and instead based on the runtime classpath, it'd be much better for developers to know about this problem at application start time, not later (i.e. 'fail fast').

The new Parser Builder implementation can ensure that's the case because this logic is now in the `build` implementation - likely something to be called during application startup to create an application singleton (e.g. in Spring).

Much nicer!